src/scheduler/pbs_sched.c

```diff
@@ -240,6 +249,8 @@ die(int sig)
 		}
 	}
 
+	pthread_mutex_unlock(&cleanup_lock);
```

Comment:
here too, if a thread got the lock and the process is going to die then we don't need to release the lock, do we?